pkg/envoy/cds/new_response.go

```diff
@@ -0,0 +1,106 @@
+package cds
+
+import (
+	"fmt"
+
+	mapset "github.com/deckarep/golang-set"
+	xds_cluster "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	xds_discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/golang/protobuf/ptypes"
+
+	"github.com/openservicemesh/osm/pkg/catalog"
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/featureflags"
+)
+
+// newResponse creates a new Cluster Discovery Response.
+func newResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, cfg configurator.Configurator) (*xds_discovery.DiscoveryResponse, error) {
+	svcList, err := meshCatalog.GetServicesFromEnvoyCertificate(proxy.GetCertificateCommonName())
+	if err != nil {
+		log.Error().Err(err).Msgf("Error looking up MeshService for Envoy with SerialNumber=%s on Pod with UID=%s", proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
+		return nil, err
+	}
+	// Github Issue #1575
+	proxyServiceName := svcList[0]
```

Comment:
This can be removed entirely for the newResponse

Reply:
+1 I'm exploring how to do that now